File: CSharp/CountryApi/CountryApi/Models/FunFact.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace CountryApi.Models
{
    public class FunFact
    {
        public int Id { get; set; }
        public string Fact { get; set; }
        [JsonIgnore] public Country Country { get; set; }

        public FunFact()
        {
            Id = default;
            Fact = default;
            Country = default;
        }
        public FunFact(string fact, Country country)
        {
            Id = default;
            Country = country;
            Fact = fact;
        }
        public FunFact(FunFact funFact)
        {
            Id = funFact.Id;
            Fact = funFact.Fact;
            Country = funFact.Country;
        }
    }
}

File: Java/country-api/src/main/java/com/countryapi/resources/FunFact.java
package com.countryapi.resources;

public class FunFact {
    private final int id;
    private final String fact;
    private final Country country;

    public FunFact() {
        this.id = 0;
        this.fact = "";
        this.country = null;
    }
    public FunFact(int id, String fact, Country country) {
        this.id = id;
        this.fact = fact;
        this.country = country;
    }
    public FunFact(FunFact funFact) {
        this.id = funFact.id;
        this.fact = funFact.fact;
        this.country = funFact.country;
    }

    private int getId(){
        return id;
    }
    private String getFact(){
        return fact;
    }
    private Country getCountry(){
        return country;
    }
}
